agent-control/src/agent_type/runtime_config/on_host/filesystem.rs

```diff
@@ -148,20 +267,23 @@ fn validate_file_entry_path(path: &Path) -> Result<(), Vec<String>> {
     if errors.is_empty() {
         Ok(())
     } else {
-        Err(errors)
+        Err(errors.join(", "))
     }
 }
 
 /// Makes sure the passed directory goes not traverse outside the directory where it's contained.
-/// E.g. via relative path specificers like `./../../some_path`.
+/// E.g. via relative path specifiers like `./../../some_path`.
 ///
+/// This would make files and directories "safe" to be created inside a sub-agent's dedicated
+/// directory, as they would not be able to write outside of it
+/// (tampering with other sub-agents or worse).
 /// Returns an error string if this property does not hold.
 fn check_basedir_escape_safety(path: &Path) -> Result<(), String> {
     path.components().try_for_each(|comp| match comp {
         Component::Normal(_) | Component::CurDir => Ok(()),
         // Disallow other non-supported variants like roots or prefixes
         Component::ParentDir | Component::RootDir | Component::Prefix(_) => Err(format!(
```

Comment:
cool is this also checking for ~ for instance?

Reply:
Not exactly. Since `~` is a [shell expansion](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html) if it's passed via the shell the character would be translated into the `$HOME` dir before reaching this. If somehow the literal character `~` reaches this point then **a file or directory with that name will be created** (from a `SafePath("~")`). It's not an error.